core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -750,12 +732,8 @@ private[zio] final class FiberContext[E, A](
             val childContext = childContextRef()
 
             if (childContext ne null) {
-              currentSuperviseMode match {
-                case ForkSuperviseMode.Auto =>
-                  if (exit.interrupted) childContext.interruptAs(fiberId) else childContext.await
-                case ForkSuperviseMode.Interrupt => childContext.interruptAs(fiberId)
-                case ForkSuperviseMode.Await     => childContext.await
-              }
+              val interruptors = exit.fold(_.interruptors, _ => Set.empty)
+              childContext.interruptAs(interruptors.headOption.getOrElse(fiberId))
```

Comment:
We are losing some information here because the `Exit` could contain multiple interruptors but `interruptAs` only takes a single `Fiber.Id`. If we wanted to I think we could change `interruptAs` and `kill0` to take a `Set[Fiber.Id]` and construct an `interruptedCause` in `kill0` that combines them all together.

Reply:
I think it makes sense to add `interruptAsAll` or add an overload, one for `Set`, one for a single fiber id. Then we could preserve all information.

Reply:
Okay let me do that.